File: Week5/Practical/mmult_skel.f90
program matrixmult
   use omp_lib
   implicit none
   integer (kind=4) :: i,j,k,ierr,indx
   integer (kind=4) :: seed(33)
   real (kind=4), allocatable :: rvect(:)
   type matrix 
     integer (kind=4) :: nrow, ncol
     real (kind=4), allocatable :: matrix(:,:)
   end type
   type (matrix) :: a, b, c


! Read in the matrix sizes
   write(6,*) ' Enter size of A (row,column) '
   read(5,*) a%nrow, a%ncol
   write(6,fmt='(2(a,i0))') ' Matrix A rows=',a%nrow,' columns=',a%ncol

   write(6,*) ' Enter size of B (row,column) '
   read(5,*) b%nrow, b%ncol
   write(6,fmt='(2(a,i0))') ' Matrix B rows=',b%nrow,' columns=',b%ncol


! Check values 
   if (a%ncol .NE. b%nrow) then
     write(6,*) ' These matrices cannot be multiplied, stopping'
     stop
   else 
     c%nrow = a%nrow
     c%ncol = b%ncol
   endif

! Assign space 

! Initialise matrices

! Perform A.B

! Write to a file

end program matrixmult

File: Week5/Practical/mmult_skel.c
#include <stdio.h>
#include <stdlib.h>
#include <omp.h>

struct matrix {
   int nrow;
   int ncol;
   float *matrix;
};

int getindex(struct matrix m, int i, int j);


int main(void) {
   int i,j,k;
   struct matrix a, b, c;
   FILE *fp;

/* Read in the matrix sizes */
   printf(" Enter size of A (row,column)\n");
   scanf("%d, %d",&a.nrow,&a.ncol);
   printf(" Matrix A rows=%d columns=%d\n",a.nrow,a.ncol);

   printf(" Enter size of B (row,column)\n");
   scanf("%d, %d",&b.nrow,&b.ncol);
   printf(" Matrix A rows=%d columns=%d\n",b.nrow,b.ncol);

/* Check values */
   if (a.ncol != b.nrow) {
     printf(" These matrices cannot be multiplied, stopping\n");
     exit(1);
   } else {
     c.nrow = a.nrow;
     c.ncol = b.ncol;
   }

/* Assign space */

/* Generate random arrays */

/* Perform A.B */

/* Write to a file */

}


int getindex(struct matrix m, int i, int j){

    return m.ncol*i + j;
}
 